File: 1. 30 Days Challenge/code/Day 7/src/twosum.go
package main

import "fmt"


func twoSum(nums []int, target int) []int {
   
    numToIndex := make(map[int]int)

    
    for i, num := range nums {
        
        complement := target - num

        
        if index, found := numToIndex[complement]; found {
            
            return []int{index, i}
        }

       
        numToIndex[num] = i
    }

    return []int{}
}

func main() {
    
    nums := []int{2, 7, 11, 15}
    target := 9
    result := twoSum(nums, target)
    fmt.Println(result) 
}

File: 1. 30 Days Challenge/code/Day 7/src/twosum.ts
function twoSum(nums: number[], target: number): number[] {
    
    const numToIndex: { [key: number]: number } = {};

    
    for (let i = 0; i < nums.length; i++) {
        
        const complement = target - nums[i];

        
        if (numToIndex[complement] !== undefined) {
            
            return [numToIndex[complement], i];
        }

       
        numToIndex[nums[i]] = i;
    }

    
    return [];
}


const nums = [2, 7, 11, 15];
const target = 9;
console.log(twoSum(nums, target));  
